installer/engine_installer/engine_installer.go

```
@@ -153,52 +154,64 @@ func (ei *EngineInstaller) buildCreateCommand(binaryPath string) {
-	// Connecting to localhost.
-	tr := &http.Transport{
+	client := &http.Client{Transport: &http.Transport{
 		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-	}
-	client := &http.Client{Transport: tr}
-
-	// validate vch host
-	sslTrustPayload := fmt.Sprintf("{\"hostState\":{\"id\":\"%s\",\"address\":\"https://%s\",\"customProperties\":{\"__adapterDockerType\":\"API\",\"__containerHostType\":\"VCH\"}}}", vchIP, vchIP)
-	sslTrustReq, _ := http.NewRequest(http.MethodPut, fmt.Sprintf("%s/resources/hosts?validate=true", admiral), bytes.NewBuffer([]byte(sslTrustPayload)))
-	sslTrustResp, err := client.Do(sslTrustReq)
-	if err != nil || sslTrustResp.StatusCode != http.StatusOK {
-		log.Infof("error: %v\nresponse: %v\n", err, sslTrustResp)
-		log.Infoln("Cannot add vch to Admiral.")
-		return
-	}
+	}}
 
-	// trust vch host on admiral
-	sslCert, err := ioutil.ReadAll(sslTrustResp.Body)
+	// add host to admiral
+	addHostPayload, err := json.Marshal(newAddHostPayload(vchIP))
 	if err != nil {
-		log.Infoln(err)
-		log.Infoln("Cannot add vch to Admiral.")
-		return
-	}
-	if len(sslCert) > 0 {
-		sslCertReq, _ := http.NewRequest(http.MethodPost, fmt.Sprintf("%s/config/trust-certs", admiral), bytes.NewBuffer([]byte(sslCert)))
-		sslCertResp, err := client.Do(sslCertReq)
-		if err != nil || sslCertResp.StatusCode != http.StatusOK {
-			log.Infof("error: %v\nresponse: %v\n", err, sslCertResp)
-			log.Infoln("Admiral cannot trust host certificate.")
-			return
-		}
+		log.Infof("cannot marshall add host payload: %s", err.Error())
+		return fmt.Errorf("error adding host to Admiral")
 	}
 
-	// add host to admiral
-	addHostReq, _ := http.NewRequest(http.MethodPut, fmt.Sprintf("%s/resources/hosts", admiral), bytes.NewBuffer([]byte(sslTrustPayload)))
+	addHostReq, _ := http.NewRequest(http.MethodPost, fmt.Sprintf("%s/resources/clusters", admiral), bytes.NewBuffer(addHostPayload))
+	hdr := http.Header{}
+	token, _ := ioutil.ReadFile("/etc/vmware/psc/admiral/tokens.properties")
```

ditto re: ignored error comment.